staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -478,31 +272,47 @@ func (u *Unstructured) GetDeletionTimestamp() *metav1.Time {
 
 func (u *Unstructured) SetDeletionTimestamp(timestamp *metav1.Time) {
 	if timestamp == nil {
-		u.setNestedField(nil, "metadata", "deletionTimestamp")
+		RemoveNestedField(u.Object, "metadata", "deletionTimestamp")
```

Comment:
I'd expect a similar remove on `SetCreationTimestamp`

Reply:
Creation timestamp is not optional, at least the argument of the Go func is not. So at worst, we miss a feature here on the Go level.